diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 53c18153cd7c..020650137880 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -14,6 +14,7 @@ dtb-$(CONFIG_ARCH_BCM2835) += \
 	bcm2710-rpi-3-b.dtb \
 	bcm2710-rpi-3-b-plus.dtb \
 	bcm2711-rpi-4-b.dtb \
+	bcm2711-rpi-4-srcv2.dtb \
 	bcm2711-rpi-400.dtb \
 	bcm2709-rpi-cm2.dtb \
 	bcm2710-rpi-cm3.dtb \
